### PR TITLE
Feature/ega archive/batch merge 2

### PR DESCRIPTION
**Pull request by blankdots**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

**Comment by blankdots** [comment]
Decided on doing the unit-tests in a different PR (pending) - and in the meantime proceeding with this as the integration tests seem to pass. Also a test deployment has been done here: https://github.com/NBISweden/LocalEGA-helm/pull/69

**Review by jbygdell — APPROVED** [review]
I don't really like that the names for the exchanges are hardcoded in the publishing.

File: lega/utils/amqp.py [hunk 1]
```diff
@@ -12,117 +14,200 @@
...
+            try:
+                content = json.loads(body)
+            except Exception as e:
+                LOG.error('Malformed JSON-message: %s', e, extra={'correlation_id': correlation_id})
+                LOG.error('Original message: %s', body, extra={'correlation_id': correlation_id})
+                err_msg = {
+                    'reason': 'Malformed JSON-message',
+                    'original_message': body.decode(errors='ignore')  # or str(body) ?
+                }
+                publish(err_msg, 'cega', 'files.error', correlation_id=correlation_id)
```
Comment (blankdots) [inline lega/utils/amqp.py:168]: this is hardcoded exchange

File: lega/utils/amqp.py [hunk 2]
```diff
@@ -12,117 +14,200 @@
...
+                return  # game over
+
+            # Message correctly formed
+            answer, error = work(content)  # exceptions already caught by decorator
 
             # Publish the answer
             if answer:
                 assert(to_routing)
-                publish(answer, to_channel, 'lega', to_routing, correlation_id=correlation_id)
+                publish(answer, 'lega', to_routing, correlation_id=correlation_id)
```
Comment (blankdots) [inline lega/utils/amqp.py:179]: this is hardcoded exchange

File: lega/utils/errors.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    file_id = data.get('file_id', None)  # should be there
+    if file_id:
+        set_error(file_id, cause, from_user)
+    LOG.debug('Catching error on file id: %s', file_id)
+    if from_user:  # Send to CentralEGA
+        org_msg = data.pop('org_msg', None)  # should be there
+        org_msg['reason'] = str(cause)  # str = Informal
+        LOG.info('Sending user error to local broker: %s', org_msg)
+        publish(org_msg, 'cega', 'files.error')  # will fetch itself the correlation_id
```
Comment (blankdots) [inline lega/utils/errors.py:62]: this is hardcoded exchange

**Review by viklund — APPROVED** [review]
I think Jocke is correct about the exchanges. But...

**Comment by blankdots** [comment]
merging this and will create a issue to handle the exchanges.

**Pull request merged**